app/controllers/user.js

```diff
@@ -34,34 +60,13 @@ exports.adminSignUp = (req, res, next) => {
         }
       });
     } else {
-      if (!emailValid(req.body.email)) {
-        logger.error(`Email: ${req.body.email} invalid.`);
-        return next(errors.emailNotValid(req.body.email));
-      }
-      if (!passwordValid(req.body.password)) {
-        logger.error('Password invalid.');
-        return next(errors.passwordInvalid);
-      }
-
-      const newAdmin = {
-        firstName: req.body.firstName,
-        lastName: req.body.lastName,
-        email: req.body.email,
-        password: req.body.password,
-        admin: true
-      };
-
-      logger.info(`All validations passed, going to create the newAdmin: ${JSON.stringify(newAdmin)}`);
-      bcrypt
-        .hash(newAdmin.password, saltRounds)
-        .then(hash => {
-          newAdmin.password = hash;
-          return User.createModel(newAdmin).then(auxUser => {
-            res.status(201).send({ newAdmin: auxUser });
-          });
+      generateUser(user, next);
```

Comment:
this function call will return a promise, so the user creation is being executed asynchronously and if you don't add a .then(), you won't be waiting to the execution to end. So you automatically continue with the next line and try to give the user admin params, that also runs asynchronously, so you are running concurrently the creation of the user with update to give admin privileges. 
So in some cases you might have problems like that the user doesn't exists.

One correct way of doing this will be 
```
return generateUser(userParams).then(user => {
  return giveAdminPriviledges(user).then(admin => {
    res.status(201);
    res.send({ admin });
}
```

With this way you can also use the user instance in the giveAdminPrivilege function as I told you before.

Reply:
Also the params you are defining above already contain the `admin: true` value so you don't need the `giveAdminPrivilege` call.